podman/domain/config.py

```diff
@@ -155,19 +157,128 @@ def services(self):
+        logging.debug("Podman client initialized.")
+
+        # Read Podman configuration
+        podman_config = PodmanConfig()
+        logging.debug("Podman configuration loaded.")
+
+        # Extract network options for the specified network
+        network_opts = podman_config.network_options.get(network_name, [])
+        logging.debug(f"Original network options for '{network_name}': {network_opts}")
+
+        # Append the port mapping using '-T' flag
+        network_opts += ["-T", port_mapping]
+        logging.debug(f"Updated network options for '{network_name}': {network_opts}")
+
+        # Create the container with network options
+        logging.info(f"Creating container '{container_name}' with image '{image}' on network '{network_name}' with options {network_opts}")
+
+        container = podman_client.containers.create(
+            image=image,
+            name=container_name,
+            networks={network_name: {}},
+            network_options={network_name: network_opts},
+            **kwargs  # Include other parameters like environment variables, volumes, etc.
+        )
+        logging.info(f"Container '{container_name}' created successfully.")
+
+        # Start the container
+        container.start()
+        logging.info(f"Container '{container_name}' started successfully.")
+
+        return container
+
+    except PodmanError as pe:
+        logging.error(f"Podman error occurred: {pe}")
+        raise
+    except PermissionError as pe:
+        logging.error(f"Permission error: {pe}")
+        raise
+    except Exception as e:
+        logging.error(f"An unexpected error occurred: {e}")
+        raise
+
+
+if __name__ == "__main__":
+    try:
+        # Define container parameters
+        network_name = "pasta"              # Replace with your network name if different
+        port_mapping = "3128:3128"          # Host port : Container port
+        image = "your-image"                 # Replace with your actual image
+        container_name = "your-container"    # Replace with your desired container name
+
+        # Additional parameters (if any)
+        additional_kwargs = {
+            # Example: Environment variables
+            # "environment": {"ENV_VAR": "value"},
+            # Example: Volume mounts
+            # "volumes": {"/host/path": {"bind": "/container/path", "mode": "rw"}},
+        }
+
+        # Create and start the container
+        container = create_container_with_pasta(
+            network_name=network_name,
+            port_mapping=port_mapping,
+            image=image,
+            container_name=container_name,
+            **additional_kwargs
+        )
+
+        print(f"Container '{container.name}' is running with TCP namespace forwarding from host port {port_mapping.split(':')[0]} to container port {port_mapping.split(':')[1]}.")
+
+    except Exception as e:
+        print(f"Failed to create and start container: {e}")
```

Comment:
I am perplexed about the `if __name__ == "__main__":` block. What would be it's use case?

There are circular imports failures related to it, so I would remove it. The example looks nice though so actually it could be documented.